### ISSUE TITLE
Support package.json in different folder than .git

### ISSUE THREAD

**Issue by tauren** [bahmutov/pre-git#20]
Do to a restriction outside of my control, we have a single git repo that contains multiple node projects. The repo is structured like this:

```
/.git/
/.git/hooks/
/project1/
/project1/package.json
/project1/node_modules
/project2/
/project2/package.json
/project2/node_modules
```

The `pre-git` hooks assume that there is only one `package.json` and that it is in the same directory as the `.git` folder. What would be involved in getting the hook to walk up the directory tree and locate the first `package.json`? This would allow me to run `git commit` in the `/project1` directory and find its `package.json` instead of throwing an error that the file doesn't exist.

By the way, it looks like `process.cwd()` reports the directory that contains `/.git`, however  `process.env.GIT_PREFIX` contains the sub-path to where the user ran the git command.

**Comment by bahmutov** [comment]
You would like to find package.json somewhere between the current working
folder and the repo's root folder?

**Comment by tauren** [comment]
Yes. Either in the current working directory, or a parent of it, up to the repo's root folder.

I just noticed support for `pre-git-cwd` and am messing with it.

**Comment by tauren** [comment]
`pre-git-cwd` won't work since it is a setting within `package.json`, which is what I'm trying to locate and haven't loaded yet.

**Comment by tauren** [comment]
https://github.com/indexzero/node-pkginfo is a popular package that will locate and load the `package.json` file. Since I assume the hook scripts shouldn't have dependencies, the code could be borrowed from here:
https://github.com/indexzero/node-pkginfo/blob/master/lib/pkginfo.js#L90

If I implemented this, would you accept a PR?

**Comment by bahmutov** [comment]
I think this works now, I released it as 0.2.0 and added you to the list of contributors in the package.json. Thank you for this feature.
